hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -836,4 +844,12 @@ private static void updateBodyForAutoApprovalOfNonFungible(
         // have a matching ownership change in its token transfer list
         log.error(CHANGE_SWITCHED_TO_APPROVAL_WITHOUT_MATCHING_ADJUSTMENT_IN, switchedChange, opBuilder);
     }
+
+    private boolean isSystemAccountDetected(final BalanceChange change) {
+        final var accountNum = change.counterPartyAccountId() != null
+                ? change.counterPartyAccountId().getAccountNum()
+                : change.getAccount().num();
+
+        return accountNum <= 750;
```

Comment:
This should really use `ContractsModule.provideStrictHederaSystemAccountDetector()` rather than having a magic number here.

Reply:
I agree with you that we need to get rid of the magic number. However, instead of using the `ContractsModule.provideStrictHederaSystemAccountDetector()` predicate, which accepts an address type, I simply created a constant `ContractsModule.SYSTEM_ACCOUNT_BOUNDARY`  to avoid back-and-forth conversions from accountIds to addresses. Does this seem reasonable to you?

Reply:
FIne, but: there are _two_ boundaries we use: strict at 750 and "regular" (I guess?) at 1000.  Maybe provide _both_ constants here even though you're only using one now.

Reply:
Done, provided const for the strict boundary also.